internal/driver/driver.go

```diff
@@ -1,36 +1,73 @@
 package driver
 
 import (
+	"errors"
 	"log"
 
-	"github.com/seemsod1/api-project/internal/config"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/seemsod1/api-project/internal/models"
 
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
 )
 
-// DB is a struct that embeds the gorm.DB
-type DB struct {
+// DatabaseConfig is a struct that holds the database configuration
+type DatabaseConfig struct {
+	DSN string
+}
+
+func NewDatabaseConfig() *DatabaseConfig {
+	var dbConfig DatabaseConfig
+	if err := envconfig.Process("DB", &dbConfig); err != nil {
+		log.Fatal(err)
+	}
+	return &dbConfig
+}
+
+func (dc *DatabaseConfig) Validate() bool {
+	return dc.DSN != ""
+}
+
+// GORMDriver is a struct that embeds the gorm.DB
+type GORMDriver struct {
 	SQL *gorm.DB
 }
 
-var dbConn = &DB{}
+func NewGORMDriver() *GORMDriver {
+	return &GORMDriver{}
+}
+
+// ConnectSQL connects to the database
+func (gd *GORMDriver) ConnectSQL() (*GORMDriver, error) {
+	cfg := NewDatabaseConfig()
+
+	if !cfg.Validate() {
+		return nil, errors.New("invalid database configuration")
+	}
+
+	d := gd.openDB(cfg)
+
+	dbConn := &GORMDriver{}
+
+	dbConn.SQL = d
+
+	return dbConn, nil
+}
 
 // openDB opens a database connection
-func openDB(env *config.EnvVariables) *gorm.DB {
-	db, err := gorm.Open(postgres.Open(env.DSN), &gorm.Config{})
+func (gd *GORMDriver) openDB(cfg *DatabaseConfig) *gorm.DB {
+	db, err := gorm.Open(postgres.Open(cfg.DSN), &gorm.Config{})
 	if err != nil {
 		log.Fatal("error connecting to database: ", err)
 	}
 
 	return db
 }
 
-// ConnectSQL connects to the database
-func ConnectSQL(env *config.EnvVariables) (*DB, error) {
-	d := openDB(env)
-
-	dbConn.SQL = d
-
-	return dbConn, nil
+func (gd *GORMDriver) RunMigrations() error {
+	err := gd.SQL.AutoMigrate(&models.Subscriber{})
```

Comment:
I am not sure whether it is a good idea to move the application-specific migrations with hardcoded values (like `models.Subscriber{}`) to the driver package. Consider moving them back to `setup.go` instead, since the current implementation tightly couples the database layer with the specific schema details of the application

Reply:
I do not know exactly where I really need to store migrations, because in package ```storage``` I hold functions to work with the database, and in the driver, I work with database connections. So it's a little bit confusing.